ghdl: add support for llvm backend

timor opened the pull request:
###### Motivation for this change

This adds support for building ghdl with the llvm backend, which enables among other things, to
have a working version for 64 bit systems.  Unfortunately it only works in 16.03, not master, since
gnat is broken there.
###### Things done

Added new attribute "ghdl_llvm" in addition to "ghdl_mcode", by introducing the argument "flavour"
to the ghdl nix expression and choosing the build flags according to the specified flavour.
- [ ] Tested using sandboxing
  ([nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS,
    or option `build-use-chroot` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Make the existing ghdl recipe more flexible, and introduce "ghdl_llvm"
as a package in addition to "ghdl_mcode".  This seems to specifically
require llvm 3.5, though.

cc @viric

offlinehacker committed:
runc: init at 2016-06-15

offlinehacker committed:
containerd: init at 0.2.3

timor committed:
ghdl: add support for llvm backend

Make the existing ghdl recipe more flexible, and introduce "ghdl_llvm"
as a package in addition to "ghdl_mcode".  This seems to specifically
require llvm 3.5, though.

cc @viric

veprbl committed:
root: add support for x86_64-darwin

1) add pcre dependency (for some reason builtin_pcre doesn't work)
2) Disable dependencies that are currently not supported by the
expression.  Most users should not need those. These are disabled to
prevent cmake from picking them up from system and causing impurities.
Once there is a user who needs these they will have to update the
expression.
3) disable some OSX detection code that relies on /usr/bin/sw_vers
that chooses c++ library, silences warnings and sets macosx-version-min.
macosx-version-min is already set by nix using MACOSX_DEPLOYMENT_TARGET
environment variable.

vbgl committed:
lua-sockets: fix on darwin

vbgl committed:
mpv: fix on darwin

abbradar committed:
linuxPackages.displaylink: init at 1.1.62

abbradar committed:
displaylink service: init

kamilchm committed:
python27Packages.cairocffi: FIX test_scaled_font

workaround for https://github.com/Kozea/cairocffi/issues/88

(cherry picked from commit 9e18b81dedec6d06f5bdfc80c26f281f1b07a2d1)
Signed-off-by: Domen Kožar <domen@dev.si>

dezgeg committed:
lib.chooseDevOutputs: Use lib.getDev

Reduces duplication, plus is actually needed for Go packages (at least
go-repo-root).

joachifm commented on pkgs/development/compilers/ghdl/default.nix:
```diff
@@ -1,21 +1,25 @@
+assert flavour == "llvm" || flavour == "mcode";
 
 let
+  optional = stdenv.lib.optional;
```
Equivalently, `inherit (stdenv.lib) optional`

joachifm commented on pkgs/development/compilers/ghdl/default.nix:
```diff
@@ -1,21 +1,25 @@
 in
 stdenv.mkDerivation rec {
-  name = "ghdl-mcode-${version}";
+  name = "ghdl-${version}";
```
In keeping with the original name, maybe embed the flavour in the name?

joachifm commented on pkgs/development/compilers/ghdl/default.nix:
```diff
@@ -1,21 +1,25 @@
-  name = "ghdl-mcode-${version}";
+  name = "ghdl-${version}";
 
   src = fetchurl {
```
While you're at it, maybe port this to `fetchFromGitHub`, it should be more stable when using github's generated tarballs.

joachifm commented on pkgs/development/compilers/ghdl/default.nix:
```diff
@@ -1,21 +1,25 @@
   };
 
-  buildInputs = [ gnat zlib ];
+  buildInputs = [ gnat zlib ] ++ optional (flavour == "llvm") [ clang ncurses ];
```
I think you want `optionals` here; `optional cond x == [ x ]` if `cond` is true.

chris-martin committed:
steam: runtime-generated.nix md5->sha256

bjornfor committed:
android-studio: enable Android emulator support

These changes are needed to be able to run the system emulator (QEMU)
from Android Studio. In addition to the added dependencies,
$LD_LIBRARY_PATH had to be changed from --set to --prefix, so that libGL
is found (on NixOS).

peti committed:
configuration-hackage2nix.yaml: disable broken builds

danbst committed:
tomcat: cleanup package with callPackages

danbst committed:
tomcat: split default webapps to separate output (~6M)

danbst committed:
tomcat service: call shutdown in preStop, because postStop is too late (systemd kills process)

danbst committed:
tomcat: 7.0.68 -> 7.0.70, 8.0.32 -> 8.0.37, init 8.5.5, init 9.0.0.M10

chris-martin committed:
haskell-hasktags: don't install the "test" binary

Closes https://github.com/NixOS/nixpkgs/pull/18403.

Commit:
eclipse-plugin-rustdt: init at 0.6.2

[Bjørn: sort alphabetially in plugins.nix, capitalize meta.description,
add space around assignment operator, indent multi-line string.]

danbst committed:
tomcat: add danbst as maintainer

rzetterberg committed:
slack: 2.1.0 -> 2.1.2

peti committed:
haste-cabal-install: mark as broken

The build depends on the broken haste-Cabal package.

peti committed:
haste-compiler: mark as broken

  Configuring haste-compiler-0.5.5.0...
  Setup: At least the following dependencies are missing:
  HTTP -any,
  bzlib -any,
  either -any,
  ghc-simple -any,
  system-fileio -any,
  tar -any

peti committed:
jhc: 32 bit builds are broken in "vector"

http://hydra.nixos.org/build/40060997/nixlog/1/raw

peti committed:
git-annex: fix GHC 8.x build

peti committed:
cryptol: switch default build to GHC 8.x

peti committed:
darcs: switch default build to GHC 8.x

peti committed:
hackage-packages.nix: update Haskell package set

This update was generated by hackage2nix v2.0.1-7-g2231107 using the following inputs:

  - Hackage: https://github.com/commercialhaskell/all-cabal-hashes/commit/3d791385a6c2a1aeb769e3f73193ee5fa068d545
  - LTS Haskell: https://github.com/fpco/lts-haskell/commit/56135ef31ada6f0ad74b1ffbf769a369f31fd2ef
  - Stackage Nightly: https://github.com/fpco/stackage-nightly/commit/a8a4229c983aa8737495a57f37b77805efdb7c57

peti committed:
git-annex: update version to 6.20160907

peti committed:
haskell-servant-router: don't run the test suite

It times out after 2 hours: http://hydra.nixos.org/build/40065332.

(cherry picked from commit 277efdd50eeec48a510672eaea5d7d664aac0b7c)

bendlas committed:
wine: 1.8.3 -> 1.8.4; 1.9.16 -> 1.9.18

- update mono and gecko versions
- update winetricks
- replace sourceforge download urls

dezgeg committed:
util-linux: Split 'dev'

dezgeg committed:
udev: Remove hacky alias

It's not clear to me what this is achieving, plus for some reason this
is causing an evaluation error in hyperterm. So let's hope it's not
really needed...

joachifm committed:
mpd service: add types to all options

joachifm committed:
mpd service: replace script with serviceConfig.ExecStart

danbst committed:
tomcat service: bump default tomcat to 8.5

See migration changelogs at
- 7.0 -> 8.0: https://tomcat.apache.org/migration-8.html
- 8.0 -> 8.5: https://tomcat.apache.org/migration-85.html

copumpkin committed:
twelf: fix emacs mode

dezgeg committed:
python-systemd: Fix build

mattbillenstein committed:
mongodb: 3.2.1 -> 3.2.9

[Bjørn: add missing hash update.]

rasendubi committed:
cmocka: fix build for Darwin (#18448)

See https://open.cryptomilk.org/issues/43 for upstream issue.

cleverca22 committed:
toxvpn: 20160606 -> 2016-09-09

Closes #18459

dezgeg committed:
Merge remote-tracking branch 'upstream/staging' into master

the-kenny committed:
mars: Fix build caused by missing GL/glu.h import

ikervagyok committed:
Linux-kernel: Workaround for https://github.com/NixOS/nixpkgs/issues/18451

remove after upstream gets fixed

shlevy committed:
Merge branch 'test' of https://github.com/ikervagyok/nixpkgs

Linux-kernel: Workaround for keyboard not working (#18462)

peti committed:
Cosmetic, no functional change.

peti committed:
haskell-lens: fix build on older compilers

romildo committed:
feh: 2.16.1 -> 2.17.1

romildo committed:
feh: perlPackages needed only at build time

peti committed:
r-XML: fix build

peti committed:
hackage-packages.nix: update Haskell package set

This update was generated by hackage2nix v2.0.1-8-g914b77b using the following inputs:

  - Hackage: https://github.com/commercialhaskell/all-cabal-hashes/commit/aa7348b0fd9e10d5d339f6c20bcdd2ab61ad64f2
  - LTS Haskell: https://github.com/fpco/lts-haskell/commit/56135ef31ada6f0ad74b1ffbf769a369f31fd2ef
  - Stackage Nightly: https://github.com/fpco/stackage-nightly/commit/8b7d8b236d0c477eda978deba79f0f71cc02916d

romildo committed:
feh: avoid using xlibsWrapper

peti committed:
r-curl: fix build

peti committed:
r-openssl: fix build

domenkozar committed:
pythonPackages.libtmux: fix build

(cherry picked from commit ac46759f7a0d6ce49a34984df5fa1fd8c6dc6c91)
Signed-off-by: Domen Kožar <domen@dev.si>

domenkozar committed:
pypeg2: 2.15.1 -> 2.15.2 (fix build)

(cherry picked from commit ace7bbb06c966f1c4c4dfa96ebfc0eeffee4874d)
Signed-off-by: Domen Kožar <domen@dev.si>

lsix committed:
libyaml: 0.1.6 -> 0.1.7

lsix committed:
pythonPackages.pyyaml: 3.11 -> 3.12

domenkozar committed:
renpy: fix build after multiple outputs

(cherry picked from commit d56bf316556877bf26bf2c2a4b83dcff3ecae161)
Signed-off-by: Domen Kožar <domen@dev.si>

ttuegel committed:
nixos/tests/sddm-kde5: don't run tests through krunner

(cherry picked from commit 1f510dc7cd1dbb67bbf6574983de1dc5ebde1a1d)

ttuegel committed:
nixos/tests/kde5: rename from sddm-kde5 and run by default

(cherry picked from commit 701f02a6eed902835e3d2a27838723cd9a7bb66a)

joachifm committed:
Merge pull request #18317 from Profpatsch/mustache-docs

mustache: init mustache at 1.0.2

mdorman committed:
Fix borgbackup dependency on lz4

roconnor committed:
duplicity: add dependency on rsync

When using the rsync:// protocol, duplicity expects to find the rsync binary in the path.
Without rsync in the path, duplicity fails with the following error

    Attempt 1 failed. AttributeError: 'NoneType' object has no attribute 'rfind'

Adding rsync to the path enables the rsync:// protocol to work correctly.

qknight committed:
Merge pull request #18469 from mdorman/borgbackup-fix

Fix borgbackup dependency on lz4

domenkozar committed:
Remove docker-registry as it's deprecated #18209

edolstra committed:
runInMachine: Unset another variable

Thanks @cstrahan.

vcunat committed:
Merge #18434: slack: 2.1.0 -> 2.1.2

rycee committed:
svtplay-dl: 1.1 -> 1.3

fpletz committed:
multipath-tools: 0.5.0 -> 0.6.2, fixes build

Was broken due to 78178d5854901e1b17a14bce3fe43515984b7b91.

vcunat committed:
gcc45, gnat, ghdl: fix up the builds

Some parts are slightly puzzling, but it seems to work and it didn't
seem economical to put more effort into it.

offlinehacker committed:
go-md2man: init at 1.0.6

vcunat committed:
gccgo-4.8: remove

I see no use to keep it. I doesn't build since April,
and noone has bothered to fixup the multiple-output problem.

vcunat committed:
pythonPackages.qscintilla: fixup build

It's the same as fcc76325ef2.

michalrus committed:
beets: add copyartifacts plugin

michalrus committed:
beets: fix keyfinder plugin by addind keyfinder-cli dependency

ttuegel committed:
openblas: 0.2.17 -> 0.2.19

domenkozar committed:
Remove test leftover from docker-registry

(cherry picked from commit d171c59926b77564089964f06ba761609b9a477e)
Signed-off-by: Domen Kožar <domen@dev.si>

vcunat committed:
broadcom-sta: don't (claim to) maintain anymore

I haven't used the driver for years.

FRidh committed:
Merge pull request #18332 from offlinehacker/pkgs/pythonPackages/docker_compose/update/1.8.0

pythonPackages.docker_compose: 1.6.2 -> 1.8.0

vcunat committed:
libav: 11.7 -> 11.8

vcunat committed:
Merge branch 'master' into staging

FRidh committed:
Merge pull request #18025 from knedlsepp/fix-python3.5-jupyter-1.0.0

python3.5-jupyter: Fix build issue #17902

vcunat committed:
orc: bug-fix 0.4.25 -> 0.4.26

vcunat committed:
ffmpeg_2: 2.8.6 -> 2.8.7

vcunat committed:
gtk2: maintenance 2.24.30 -> 2.24.31

vcunat committed:
pango: maintenance 1.40.1 -> 1.40.2

vcunat committed:
Merge #18400: shared-mime-info: 1.5 -> 1.7

jgeerds committed:
pytest-rerunfailures: init at 2.0.1

(cherry picked from commit 6ddf8f31d3998e464c10fdb59385bf79cf08ef09)
(cherry picked from commit 074013bca596571a33852f18ff0f29c4578c347c)

vcunat committed:
mesa: maintenance 12.0.1 -> 12.0.2

vcunat committed:
Merge #16391: texlive: 2015 -> 2016

Mirroring isn't finalized, but we'll have to fix that on the go.

abuibrahim committed:
pktgen: 3.0.04 -> 3.0.13 (#18477)

rycee committed:
udiskie: move to all-packages

This is an application, not a python library and should therefore be in
its own package.

jgeerds committed:
tmuxp: Add pytest-rerunfailures to dependencies

(cherry picked from commit 6fc67944bf8a9841b649c8d36f852ce65bbae0e1)
(cherry picked from commit f0d1ac606232a02771343f6ad84d4c769ae2ad54)

peti committed:
r-openssl: fix build some more

offlinehacker committed:
docker: 1.10.3 -> 1.12.1

offlinehacker committed:
docker module: updates

- logDriver option, use journald for logging by default
- keep storage driver intact by default, as docker has sane defaults
- do not choose storage driver in tests, docker will choose by itself
- use dockerd binary as "docker daemon" command is deprecated and will be
  removed
- add overlay2 to list of storage drivers

peti committed:
net-snmp: cosmetic fix for syntax highlighting in Emacs

vcunat committed:
texlive: add links to ht* tools, in particular htlatex

This should fix #18480.

nbp committed:
Fix #18316 - Revert #14398, Using self prevent aliasing overriden packages.

joachifm committed:
Merge pull request #18453 from chris-martin/atermjava

atermjava: remove

joachifm committed:
grsecurity: 4.7.2-201608312326 -> 4.7.3-201609072139

joachifm committed:
Merge pull request #18446 from siddharthist/docs/ipv6-per-interface

nixos manual: disable ipv6 per interface

shlevy committed:
melpa: Add inherit-local, bump f.el

joachifm committed:
purePackages.glpk: fix build

Add gmp.dev to the include path to help glpk find gmp.h.

https://github.com/NixOS/nixpkgs/issues/18209

joachifm committed:
Merge pull request #17781 from knedlsepp/fix-armadillo-build

armadillo: Revert removal of LAPACK patch.

groxxda committed:
treewide: deprecate ip-up.target (#18319)

Systemd upstream provides targets for networking. This also includes a target network-online.target.

In this PR I remove / replace most occurrences since some of them were even wrong and could delay startup.

dezgeg committed:
e2fsprogs: Split 'dev'

langston-barrett committed:
ympd service: init (#18371)

ympd provides a web ui, it is suitable to be run as a service.
Fixes #17878.

service has no requirements b/c user might be using remote mpd
instance.

FRidh committed:
julia: mark as broken for i686

FRidh committed:
julia-git: mark as broken for i686

aske committed:
linuxPackages.mba6x_bl: 2016-02-12 -> 2016-04-22

edwtjo committed:
idea.idea-{community,ultimate}: 2016.2.3 -> 2016.2.4

edwtjo committed:
idea.idea14-ultimate: use base32 hash

chris-martin committed:
cua-mode: md5->sha256

chris-martin committed:
batik: md5->sha256

chris-martin committed:
docbook-ebnf: md5->sha256

chris-martin committed:
docbook 4.3: md5->sha256

chris-martin committed:
scilab: md5->sha256

chris-martin committed:
kdesvn: md5->sha256

chris-martin committed:
ocaml 3.08.0: md5->sha256

chris-martin committed:
lua-4: md5->sha256

chris-martin committed:
pythonPackages.breathe: md5->sha256

chris-martin committed:
pythonPackages.rhpl: md5->sha256

chris-martin committed:
gnumake-3.80: md5->sha256

chris-martin committed:
mk: md5->sha256

peti committed:
haskell-yi: remove obsolete "broken" marker

Fixes https://github.com/NixOS/nixpkgs/issues/18188.

peti committed:
haskell-gtk2hs-buildtools: this override does not belong into common

peti committed:
haskell-configuration-common: cosmetic

aske committed:
mpv: 0.19.0 -> 0.20.0 (#18493)

Enable vaapiSupport by default.

peti committed:
idris: clean up overrides

Closes https://github.com/NixOS/nixpkgs/pull/18309.

peti committed:
Merge pull request #17145 from kalhauge/haskell_darwin_dyld

haskell-ghc: fix darwin issue

peti committed:
Merge pull request #17997 from teozkr/fix/buildstackproject-stackroot

buildStackProject: Fix missing STACK_ROOT causing build errors

dezgeg committed:
Merge commit 'bd6e40c' from staging into master

Major changes being pulled in:

- mesa: maintenance 12.0.1 -> 12.0.2
- texlive: 2015 -> 2016

fpletz committed:
libopus: 1.1.2 -> 1.1.3

fpletz committed:
mpd: 0.19.15 -> 0.19.19

aneeshusa committed:
cryptol: Make Z3 available for REPL users

The Cryptol REPL has a hard dependency on Z3, but the rest of the
library uses SBV to support multiple solvers. Ensure that Z3 is
available for `pkgs.cryptol`, which is likely to be installed via
nix-env for REPL usage, but do not change pkgs.haskellPackages.cryptol,
which is likely to be used as a dependency (in Nix expressions).

dezgeg committed:
vlc_npapi: Fix hash

dezgeg committed:
release-lib: Remove unused allowTexliveBuilds option

This is dead code since the old monolithic TeXLive was dropped in
da421bc75f98c1b19f214a3b6b7cda07dc4c088b.

peti committed:
Merge pull request #18497 from aneeshusa/add-z3-dependency-for-cryptol

cryptol: Make Z3 available for REPL users

Commit:
cabin: init at 1.005

Commit:
dosis: init at 1.007

Commit:
encode-sans: init at 1.002

Commit:
libre-baskerville: init at 1.000

Commit:
libre-bodoni: init at 2.000

Commit:
libre-caslon: init at 1.002

Commit:
libre-franklin: init at 1.014

fpletz committed:
treewide: fix fallout of ip-up deprecation

See #18319 for details. Starting network-online.target manually does not
work as it hangs indefinitely.

Additionally, don't treat avahi and dhcpcd special and sync their systemd units
with the respective upstream suggestion.

fpletz committed:
toxvpn service: doesn't require online network

Tested that it detects network changes quickly.

vcunat committed:
llvmPackages: generalize "3.9.0" string occurrences

Discussed:
https://github.com/NixOS/nixpkgs/commit/15af9082eb#commitcomment-18903919

dezgeg committed:
Merge remote-tracking branch 'upstream/staging' into master

vcunat committed:
libarchive: include XAR support by default

Discussion:
https://github.com/NixOS/nixpkgs/pull/17834

vcunat committed:
Merge #17834: libarchive: make XAR support configurable

vcunat committed:
libarchive: use pkgconfig

It tends to make builds more reliable, in general.

dezgeg committed:
Merge remote-tracking branch 'upstream/master' into staging

aszlig committed:
nixos/modules/rename: Remove docker-registry

This is a follow-up to 9c1cdedcba2fd5233c71f0988d2eb725cc8f32ad and
fed3501b0722e187284f9f6e1532f5b6e0572d6e.

Discussion:

https://github.com/NixOS/nixpkgs/issues/18209#issuecomment-245968857

Signed-off-by: aszlig <aszlig@redmoonstudios.org>
Cc: @domenkozar
Issue: #18209

Baughn committed:
steam: Some games use Mono

abbradar committed:
Merge pull request #18514 from Baughn/steam

steam: Some games use Mono

aszlig committed:
nixos/lib/testing: Fix unsetting $xchg

Regression introduced by 4dcb685af940efd74a7b2b66ae917129ef232d83.

Unsetting the environment variable shortly before using it is not going
to end up very well, so let's just filter out the variable from the
output of export and unset it shortly afterwards.

This fixes the runInMachine NixOS test.

Signed-off-by: aszlig <aszlig@redmoonstudios.org>

vcunat commented:
Note: gnat got fixed for unstable/master and 16.09 (see 36ef583e47).

edwtjo committed:
android-studio: 2.1.2.0 -> 2.1.3.0

domenkozar committed:
qgis: fix build

(cherry picked from commit dccf8c5f27ed4a1af47f1fc052e9ac206f3a61b7)
Signed-off-by: Domen Kožar <domen@dev.si>

fadenb committed:
sonarr: 2.0.0.4230 -> 2.0.0.4323

fadenb committed:
emby: 3.0.6400 -> 3.0.7100

peti committed:
hackage-packages.nix: update Haskell package set

This update was generated by hackage2nix v2.0.1-8-g914b77b using the following inputs:

  - Hackage: https://github.com/commercialhaskell/all-cabal-hashes/commit/3a577eda9a11923156fe4f3006ca914673c3ffde
  - LTS Haskell: https://github.com/fpco/lts-haskell/commit/36c0f4fa5e01617466cc1de3449e52e8c881eead
  - Stackage Nightly: https://github.com/fpco/stackage-nightly/commit/8b258a761d580f7be233d4d44d892d27d5240ccf

fpletz committed:
Merge pull request #18298 from teh/prometheus-service

Prometheus service

fpletz committed:
prometheus service: move to separate folder

aske committed:
treewide: explicitly specify gtk and related package versions

aske committed:
gnome2.at_spi: disable hardening to fix build

aske committed:
gnome2.gnome_session: mark as broken

aske committed:
gtkmathview: mark as broken

aske committed:
nixos manual: mention gtk-related alias changes

aske committed:
gnome2.gnome_settings_daemon: add libSM to fix build

aske committed:
gnome2.gnome_control_center: add libSM to fix build

joachifm committed:
Merge pull request #18373 from tavyc/mc

mc: Enable support for SFTP and SMB.

joachifm committed:
Merge pull request #18519 from mayflower/emby_7100

emby: 3.0.6400 -> 3.0.7100

joachifm committed:
Merge pull request #18518 from mayflower/sonarr_4323

sonarr: 2.0.0.4230 -> 2.0.0.4323

langston-barrett committed:
audio services: use mkEnableOption (#18524)

copumpkin committed:
mention-bot: notify me when darwin stdenv stuff changes

peti committed:
structured-haskell-mode: link Haskell libraries statically into this executable

It's my understanding that Emacs runs the "structured-haskell-mode" binary
virtually every time you press a key in an Haskell buffer, and since
dynamically linked Haskell binaries take *much* longer to start up, switching
this particular package to statically linked libraries ought to result in a
performance boost.

hakuch committed:
opam, aspcud: init packages for external solver (#16938)

The opam package manager relies on external solvers to determine package
management decisions it makes related to upgrades, new installations,
etc.

While, strictly speaking, an external solver is optional, aspcud is
highly recommended in documentation. Furthermore, even having a
relatively small number of packages installed quickly causes the limits
of the interal solver to be reached (before it times out).

Aspcud itself depends on two programs from the same suite: gringo, and
clasp.

On Darwin, Boost 1.55 (and thus Gringo) do not build, so we only support
Aspcud on non-Darwin platforms.

FRidh committed:
Merge pull request #18396 from rycee/fix/udiskie

udiskie: 1.4.8 -> 1.5.1

peti committed:
haskell-timezone-series: block version 1.6 update

The new version's build is completely broken: https://github.com/ygale/timezone-series/issues/2.

Commit:
matrix-synapse: 0.17.1 -> 0.17.2

shlevy committed:
melpa: Fix nix-buffer and inherit-local, add stable

dezgeg committed:
Merge commit 'e0b5eef' from staging into master

exi committed:
Fix kmymoney test build failure on 32bit

(cherry picked from commit f4618e14b956055da3df400fa9d302c5ea64a7f8)
Signed-off-by: Domen Kožar <domen@dev.si>

7c6f434c committed:
libreoffice: generate-libreoffice-srcs.sh: take into account that LO now sometimes lets the checksum go after the tarball

7c6f434c committed:
libreoffice-fresh: 5.2.0.4 -> 5.2.1.2

7c6f434c committed:
firejail: 0.9.42-rc1 -> 0.9.42

7c6f434c committed:
wine: 1.8.3 -> 1.8.4; wineUnstable: 1.9.16 -> 1.9.18

domenkozar committed:
Merge pull request #18483 from nbp/revert-14398

Using self prevent aliasing overriden packages

FRidh committed:
mendeley: 1.16.2 -> 1.16.3

vbgl committed:
Merge pull request #17973 from vbgl/mpv-darwin

mpv: fix on darwin

robgssp committed:
freerdpUnstable: 1.2.0 -> 2.0

markus1189 committed:
ntl: init at 9.11.0 (#18507)

vcunat committed:
Merge #17861: root: add support for x86_64-darwin

aristidb committed:
ispc: attempt to make build more robust and hydra-compatible

exi committed:
mumble: fix failing vm tests

modify tests to not fail if the event handlers are
registered too slowly or if the wrong window is in focus

(cherry picked from commit e087b0d12f97604ee1fdd09ef3d78b772c12468e)
Signed-off-by: Domen Kožar <domen@dev.si>

edolstra committed:
nix: Enable install check

Looks like this got disabled accidentally in
ec5b66eb4add9d494d8fb16f6899028750d317a2.

abbradar committed:
Merge branch 'displaylink'

Close #18041

dezgeg committed:
kernel generate-config.pl: Be more verbose on errors

michalrus committed:
squishyball: init at 19580 (#18354)

sternenseemann committed:
bup: 0.26 -> 0.28.1 (#18531)

cstrahan committed:
redis-desktop-manager: fix build (#18543)

We need to run the pre/post configure hooks.

ericsagnes committed:
NixOS manual: add module option types doc (#18525)

derchrisuk committed:
hhvm: 3.12.1 -> 3.15.0 (#18508)

laMudri committed:
asunder: init at 2.8 (#18510)

rycee committed:
haveged module: clean up service configuration (#18513)

Switches from the forking service type to simple by running haveged in
the foreground. Also restricts the execution environment a bit (these
are inspired by the Debian service file).

lsix committed:
pythonPackages.Mako: 1.0.2 -> 1.0.4 (#18494)

aske committed:
xflux-gui: 1.1.1 -> 2016-08-08 (#18489)

Also make it work with new python changes.

fpletz committed:
Merge pull request #18488 from chris-martin/md5-2

Miscellaneous packages: Use sha256 instead of md5 (part 2)

fpletz committed:
Merge pull request #18486 from aske/mba6x_bl

linuxPackages.mba6x_bl: 2016-02-12 -> 2016-04-22

erikryb committed:
neomutt: 20160827 -> 20160910 (#18485)

peti committed:
configuration-hackage2nix.yaml: fix evaluation errors

uralbash committed:
vscode: 1.4.0 -> 1.5.1 (#18468)

fpletz committed:
Merge pull request #18467 from lsix/update_pyyaml

Update libyaml and pyyaml

peti committed:
haskell-timezone-series: clean up overrides

groxxda committed:
networking.dhcpcd: Don't add to system closure when using networkd (#18436)

danbst committed:
nixos: fix typo in networking.interfaces.<name?>.virtual (#18548)

fpletz committed:
Merge pull request #18476 from michalrus/beets-keyfinder-copyartifacts

Beets: fix keyfinder plugin (did not work) and add copyartifacts plugin

Commit:
libspatialite: builds on unix when libiconv is included

domenkozar committed:
pypeg2: disable tests on py3k

(cherry picked from commit 2adbd13f9561edf7771fc8687b9299a817ee71b6)
Signed-off-by: Domen Kožar <domen@dev.si>

domenkozar committed:
pijul: build only on 64bit platforms

(cherry picked from commit 8479962862563783ce216c1aec1aa08c1da696b2)
Signed-off-by: Domen Kožar <domen@dev.si>

domenkozar committed:
xburst-tools: doesn't build on 32bit linux

(cherry picked from commit 635b48d373bdf98324d696419c53a81512038bb4)
Signed-off-by: Domen Kožar <domen@dev.si>

domenkozar committed:
fix eval

abbradar committed:
steam: don't write runtime tag file

abbradar committed:
Merge commit 'refs/pull/18498/head' of git://github.com/NixOS/nixpkgs

abbradar committed:
asunder: use gtk2 explicitly

abbradar committed:
banshee: use gnome2 explicitly

abbradar committed:
batti: fix eval

abbradar committed:
haskell: replace gnome mentions with gnome2 as a workaround to fix eval

Commit:
freerdp: add missing libXrender dependency

dezgeg committed:
kernel config: Explicitly enable some NLS-related things

Doesn't affect x86, but ARM can't mount VFAT filesystems without this on
a 3.18 kernel.

dezgeg committed:
kernel generate-config.pl: Properly support string options

Or we get something like:

option not set correctly: NLS_DEFAULT (wanted 'utf8', got '"utf8"')

shlevy committed:
Merge branch 'freerdp2' of git://github.com/peterhoeg/nixpkgs

CMCDragonkai committed:
sudo: Allow root to use sudo to switch groups

offlinehacker committed:
Merge pull request #18481 from offlinehacker/pkgs/docker/1.12.1

docker: 1.10.3 -> 1.12.1

vcunat committed:
gperftools: 2.4 -> 2.5

Also, maintain the package.

domenkozar committed:
nova: more transient errored test cases

(cherry picked from commit eb25ce7fbfd92070d75b06060c98b221ac8da290)
Signed-off-by: Domen Kožar <domen@dev.si>

fpletz committed:
libtorrentRasterbar: 1.1 -> 1.1.1

fpletz committed:
libtorrentRasterbar_0_16: remove old, insecure version

domenkozar committed:
fstar: disable tests https://github.com/FStarLang/FStar/issues/676

(cherry picked from commit feef017564c292e088de69e97db4071bf08c2853)
Signed-off-by: Domen Kožar <domen@dev.si>

fpletz committed:
libtorrentRasterbar_1_0: 1.0.9 -> 1.0.10

domenkozar committed:
Merge pull request #18560 from MatrixAI/root-sudo-group-switch

sudo: Allow root to use sudo to switch groups

vbgl committed:
mpd: fix on darwin

fpletz committed:
qbittorrent: 3.3.5 -> 3.3.7

aske committed:
deluge: fix libtorrentRasterbar_1_0 reference (#18564)

timor committed:
use fetchFromGitHub

for some reason the sha256 changed...

timor committed:
encode flavour in name

timor committed:
correct usage of `optional` and `optionals`

timor commented:
Works against master now, updated pull request: #18566
